### PR TITLE
ci: use same source everywhere

### PR DESCRIPTION
**Pull request by aluzzardi**
Speeds up CI considerably.

/cc @marcosnils @crazy-max 

I did see the history of this here https://github.com/dagger/dagger/pull/2288#pullrequestreview-951005783 but I didn't quite understand.

We were ending up uploading the context twice with different excludes, the other upload not excluding `node_modules` which is pretty massive

### CODE REVIEW

**Review by sipsma — APPROVED** [review]

**Comment by marcosnils** [comment]
@aluzzardi the main reason to do this is because the `Dockerfile` target doesn't need the `website` folder. If you do `dagger do build docker` without any cache, we're going to be sending the `website` folder which is quite large (~250mb) to the context unnecessarily.

**Comment by aluzzardi** [comment]
> @aluzzardi the main reason to do this is because the Dockerfile target doesn't need the website folder. If you do dagger do build docker without any cache, we're going to be sending the website folder which is quite large (~250mb) to the context unnecessarily.

The website folder without `node_modules` (which is excluded) is actually ~8MB

**Comment by marcosnils** [comment]
> The website folder without node_modules (which is excluded) is actually ~8MB

I might have missed the node_modules exclusion. I guess it's ok to merge if speeds up builds. However, having a way to exclude the website folder for the docker action would be neat. I do see this useful in a similar context with larger folder sizes

**Comment by sipsma** [comment]
I thought the issue being fixed here is just that the previous code resulted in `.` and `./` both being loaded separately, which meant that even if `./` skipped loading the website dir we weren't saving time because `.` loaded it anyways. Am I misunderstanding?

**Comment by marcosnils** [comment]
> that even if `./` skipped loading the website dir we weren't saving time because `.` loaded it anyways

The main current issue here is that the `./` loading is not excluding `node_modules` (which should be fixed) so if you do a `dagger do build` that's going to take considerably a large amount of time because of this. What we need in reality is a way to incrementally `merge` the input depending on the actions that needs it. Here's a PR which I think has the best of both worlds: https://github.com/dagger/dagger/pull/2403

WDYT?

**Comment by aluzzardi** [comment]
I don't think anything requires `website/` at the moment, I think we could just exclude it and keep a single source.

> The website folder without node_modules (which is excluded) is actually ~8MB

It's actually 3.5MB if we exclude `website/build` (which is not checked into git). Including or excluding website won't make any substantial performance difference at the end of the day. Did you notice any performance issue? Asking because I may have missed the underlying issue this is solving

**Comment by marcosnils** [comment]
> I don't think anything requires `website/` at the moment, I think we could just exclude it and keep a single source.
> 
> >

If that's the case, even better. Since it was initially there I thought there was a reason. If nothing is using it, let's just exclude it altogether and just keep a single source. :+1:

**Review by crazy-max — APPROVED** [review]

**Comment by marcosnils** [comment]
@aluzzardi let's exclude `website` before merging :pray:

**Comment by aluzzardi** [comment]
> @aluzzardi let's exclude website before merging 🙏

Done!

**Review by jlongtine — APPROVED** [review]

**Comment by marcosnils** [comment]
image


someone is back to coding :man_technologist: :tada:

**Pull request merged**
